File: lib/display/TableCell.ts
import { Empty } from "../empty_shell";

export interface TableCellProps {
    columnSpan?: number,
    rowSpan?: number,
    child: any,
    rowHover?: boolean,
    backgroundColor?: string
}

export function TableCell(id: string, {
    columnSpan = 1,
    rowSpan = 1,
    child = Empty(),
    rowHover = true,
    backgroundColor = ""
} : TableCellProps){
    return {
        id,
        name: "TableCell",
        props: {
            columnSpan,
            rowSpan,
            child,
            rowHover,
            backgroundColor
        }
    }
}

File: flutter_render/lib/src/dispaly/TableCell.dart
import 'package:neneui_render/src/enum.dart';
import 'package:shadcn_flutter/shadcn_flutter_experimental.dart';

class DTableCell {
  static TableCell run({
    required BuildContext context,
    required Map<String, dynamic> data,
    required Function reRender,
    required Function reRenderList,
    required Function event,
  }) {
    if (data['name'] == "TableCell") {
      event(Events.REGISTER_ID, {'id': data['id'], 'props': data['props']});

      return TableCell(
        columnSpan: int.parse(data['props']['columnSpan'].toString()),
        rowSpan: int.parse(data['props']['rowSpan']),
        child: reRender(data['props']['child']),
      );
    } else {
      return TableCell(child: Text(""));
    }
  }
}
